recipes/libpng/all/conanfile.py

```diff
@@ -96,7 +96,7 @@ def package_info(self):
                 self.cpp_info.libs = ['libpng16']
         else:
             self.cpp_info.libs = ["png16"]
-            if self.settings.os == "Linux":
+            if self.settings.os == "Linux" or self.settings.os == "Android":
```

Comment:
just in case, in future you may write shorter: `if str(self.settings.os) in ["Linux", "Android"]`